module/VuFind/src/VuFind/Cover/Router.php

```diff
@@ -125,23 +125,20 @@ public function getMetadata(
             return false;
         }
 
+        if (!is_array($thumb)) {
+            return ['url' => $thumb];
+        }
+
         if (!($this->config['coverimagesBrowserCache'] ?? true)) {
             // Add timestamp hash to avoid browser cache
-            $thumb['hash'] = md5(time());
+            $thumb['browser_cache_hash'] = md5(time());
         }
 
-        // Array? It's parameters to send to the cover generator:
-        if (is_array($thumb)) {
-            if (!$resolveDynamic) {
-                return null;
-            }
-            $dynamicUrl =  $this->dynamicUrl . '?' . http_build_query($thumb);
-        } else {
-            return ['url' => $thumb];
+        if (!$resolveDynamic) {
```

Comment:
Maybe it's worth keeping the previous comment in some form.

```suggestion
        // If we got this far, $thumb is an array, meaning it contains parameters to send to the cover generator.
        if (!$resolveDynamic) {
```

Reply:
Good idea, thanks!